tools/ubuntu-setup/docker.sh

```diff
@@ -14,7 +27,7 @@ sudo apt-cache policy docker-engine
 sudo apt-get -y install linux-image-extra-$(uname -r)
 
 # DOCKER
-sudo apt-get install -y --force-yes docker-engine=1.12.0-0~trusty
+sudo apt-get install -y --force-yes docker-ce
```

Comment:
We only support docker 1.12
Maybe you can add a check if ubuntu14/trusty use `docker-engine=1.12.0-0~trusty`

Reply:
Need to address the docker1.12/xenial change

Reply:
@sudswas Could you look into this and let me know what works for Xenial?

Reply:
@jonpspri If you do a apt-get install docker.io on xenial - it comes with the following:

```docker version
Client:
 Version:	17.12.0-ce
 API version:	1.35
 Go version:	go1.9.2
 Git commit:	c97c6d6
 Built:	Wed Dec 27 20:08:17 2017
 OS/Arch:	linux/ppc64le
```

Reply:
Yeah you need to override the defaults so you can get an old version of docker installed

Reply:
AFAIK, ppc64le does not have a distro/docker repo supported version of docker 1.12 - which makes it tricky.  I am not sure if the same problem exists for `s390x` as well? @jonpspri